docs/migration-v8.md

```diff
@@ -1 +1,515 @@
+    .AddRetry(new RetryStrategyOptions
+    {
+        ShouldHandle = new PredicateBuilder().Handle<Exception>(),
+        Delay = TimeSpan.FromSeconds(1),
+        MaxRetryAttempts = 3,
+        BackoffType = DelayBackoffType.Constant
+    })
+    .Build();
+
+// Synchronous execution
+pipeline.Execute(() =>
+{
+    // your code here
+});
+
+// Asynchronous execution
+await pipeline.ExecuteAsync(async cancellationToken =>
+{
+    // your code here
+},
+cancellationToken);
+
+// Create and use the ResiliencePipeline<T>
+ResiliencePipeline<HttpResponseMessage> pipelineT = new ResiliencePipelineBuilder<HttpResponseMessage>()
+    .AddRetry(new RetryStrategyOptions<HttpResponseMessage>
+    {
+        ShouldHandle = new PredicateBuilder<HttpResponseMessage>()
+            .Handle<Exception>()
+            .HandleResult(result => !result.IsSuccessStatusCode),
+        Delay = TimeSpan.FromSeconds(1),
+        MaxRetryAttempts = 3,
+        BackoffType = DelayBackoffType.Constant
+    })
+    .Build();
+
+// Synchronous execution
+pipelineT.Execute(() =>
+{
+    // your code here
+    return GetResponse();
+});
+
+// Asynchronous execution
+await pipelineT.ExecuteAsync(async cancellationToken =>
+{
+    // your code here
+    return await GetResponseAsync(cancellationToken);
+},
+cancellationToken);
+```
+<!-- endSnippet -->
+
+Here are the primary changes:
+
+- `ResiliencePipelineBuilder` is used to start building the resilience pipeline, instead of the static `Policy.HandleException<TException>(...)`.
+- For building generic resilience pipelines, `ResiliencePipelineBuilder<T>` replaces the static `Policy.HandleResult<HttpResponseMessage>(...)`.
+- Resilience strategies are configured using options, such as `RetryStrategyOptions` and `RetryStrategyOptions<T>`.
+- `PredicateBuilder` determines which exceptions or results are handled.
+- The final `ResiliencePipeline` is created with a `Build()` call.
+- `ResiliencePipeline` can execute both synchronous and asynchronous tasks.
+
+## Migrating policy wrap
+
+### Policy wrap in v7
+
+Policy wrap is used to combine multiple policies into one as shown in the v7 example below:
+
+<!-- snippet: migration-policy-wrap-v7 -->
+```cs
+IAsyncPolicy retryPolicy = Policy.Handle<Exception>().WaitAndRetryAsync(3, _ => TimeSpan.FromSeconds(1));
+
+IAsyncPolicy timeoutPolicy = Policy.TimeoutAsync(TimeSpan.FromSeconds(3));
+
+// Wrap the policies. Tne policies are executed in the following order:
+// 1. Retry
+// 2. Timeout
+IAsyncPolicy wrappedPolicy = Policy.WrapAsync(timeoutPolicy, retryPolicy);
+```
+<!-- endSnippet -->
+
+### Policy wrap in v8
+
+In v8, there's no need to use policy wrap explicitly. Instead, policy wrapping is smoothly integrated into `ResiliencePipelineBuilder`, as shown in the example below:
+
+<!-- snippet: migration-policy-wrap-v8 -->
+```cs
+// The "PolicyWrap" is integrated directly. Strategies are executed in the same order as they were added:
```

Comment:
This is exactly the comment I think is the best for understanding the changes. If I just look at the code sample, it's enough to understand what has changed from it directly and easily.